websocket.el

```diff
@@ -693,14 +693,27 @@ to the websocket protocol.
                           (port (if (= 0 (url-port url-struct))
                                     (if (eq type 'tls) 443 80)
                                   (url-port url-struct)))
-                          (host (url-host url-struct)))
+                          (host (url-host url-struct))
+                          (url-as-http (let ((url-as-http (copy-sequence url-struct)))
+                                         (setf (url-type url-as-http) (if (eq type 'plain) "http" "https"))
+                                         url-as-http))
+                          (proxy (url-generic-parse-url (url-find-proxy-for-url url-as-http (url-host url-as-http)))))
                      (if (eq type 'plain)
                          (make-network-process :name name :buffer nil :host host
```

Comment:
Will this work with http proxies?  I'd guess you would need to make the host different if it is a proxy?

Reply:
I'm not sure what you mean by "http proxies" - the code I have here only works with proxies that talk plain HTTP without encryption. (The gnutls-negotiate later on is for TLS with the websocket server)

Reply:
Sorry, I wrote something that didn't make sense.

The way I read this, it doesn't look like it would go through a proxy for unencrypted websockets.  The logic testing for proxies is only execute if `type` is not `'plain`.

Reply:
Right, this only works for wss:// at the moment. I didn't have a need for ws:// and didn't set up a testing environment for it. In theory it could be as simple as not doing the gnutls-negotiate.